LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/DefaultLoRaDataRequestHandler.cs

```diff
@@ -149,10 +152,10 @@ public async Task<LoRaDeviceRequestProcessResult> ProcessRequestAsync(LoRaReques
 
                 if (!isConfirmedResubmit)
                 {
-                    var validFcntUp = isFrameCounterFromNewlyStartedDevice || (payloadFcnt > loRaDevice.FCntUp);
+                    var validFcntUp = isFrameCounterFromNewlyStartedDevice || (payloadFcntAdjusted > loRaDevice.FCntUp);
                     if (validFcntUp)
                     {
-                        Logger.Log(loRaDevice.DevEUI, $"valid frame counter, msg: {payloadFcnt} server: {loRaDevice.FCntUp}", LogLevel.Information);
+                        Logger.Log(loRaDevice.DevEUI, $"valid frame counter, msg: {payloadFcntAdjusted} server: {loRaDevice.FCntUp}", LogLevel.Information);
```

Comment:
Wonder if we should add the original payload fcnt in log to facilitate troubleshooting.

Reply:
I do not want to pass it around everywhere. But I can add it to the log when converting it, so we have the value. In theory, it should always be inferrable from the lower 16bit.

Reply:
I am wondering if it would add a lot of value. The server side representation already carry the original payload information right? So you should be able to recover it from the value if you need it